### ISSUE TITLE
Explicit GraphQL types for Point, Line, Polygon

### ISSUE THREAD

**Issue by mattbretl** [graphile/postgis#3]
Is there a benefit to exposing all geometry/geography types as a single GraphQL type, with `__gisType` used internally as a discriminator? For web mapping purposes, you usually need to know whether you're dealing with points, lines, or polygons, so I would prefer to see GeoJSONPoint, GeoJSONLine, etc. exposed in the GraphQL schema.

**Comment by mattbretl** [comment]
Sorry, I conflated a few different issues here. This is only concerning Input types. (On the query side, you already have explicit types for each subtype, e.g. GeometryPoint.)

**Comment by benjie** [comment]
Nah, just simplicity - trivially easy to feed the GeoJSON to the DB. If they need different input types so be it 👍

**Comment by Rycochet** [comment]
Not sure if it's related (or just me doing something wrong, even if the code works) - but graphiql complains about the type on GeoJSON variables, so it's not trivial to check they're correctly formatted (I got a Polygon wrong by missing the extra nesting, and needed the backend error to notice etc):

```json
{
  "point": {
    "type": "Point",
    "coordinates": [
      -1.232,
      51.660
    ]
  }
}
```

![Screenshot 2019-03-21 at 08 57 17](https://user-images.githubusercontent.com/290714/54741799-78d78780-4bb7-11e9-9405-2d314ec1587c.png)

**Comment by benjie** [comment]
GraphiQL doesn’t like the JSON scalar either.
